octodns_constellix/__init__.py

```diff
@@ -1126,27 +1275,32 @@ def _process_desired_zone(self, desired):
             if not getattr(record, 'dynamic', False):
                 continue
 
-            if not self._is_healthcheck_configured(record):
-                incompatible_pools = []
-                for name, pool in record.dynamic.pools.items():
-                    if pool.data['fallback'] is not None:
+            incompatible_pools = []
+            for name, pool in record.dynamic.pools.items():
+                if pool.data['fallback'] is not None:
+                    incompatible = True
+                    for value in pool.data['values']:
+                        if 'obey' == value.get('status', 'obey'):
+                            incompatible = False
+                            break
+                    if incompatible:
                         incompatible_pools.append(name)
 
-                if not incompatible_pools:
-                    continue
+            if not incompatible_pools:
+                continue
 
-                incompatible_pools = ','.join(incompatible_pools)
-                msg = (
-                    f'fallback without monitor for pools '
-                    f'{incompatible_pools} in {record.fqdn}'
-                )
-                fallback = 'will ignore it'
-                self.supports_warn_or_except(msg, fallback)
+            incompatible_pools = ','.join(incompatible_pools)
+            msg = (
+                f'fallback without monitor for pools '
+                f'{incompatible_pools} in {record.fqdn}'
+            )
+            fallback = 'will ignore it'
+            self.supports_warn_or_except(msg, fallback)
 
-                record = record.copy()
-                for pool in record.dynamic.pools.values():
-                    pool.data['fallback'] = None
-                desired.add_record(record, replace=True)
+            record = record.copy()
+            for pool in record.dynamic.pools.values():
+                pool.data['fallback'] = None
+            desired.add_record(record, replace=True)
```

Comment:
What is this for? Looks pretty drastic to reset fallback for every single pool of all dynamic records.

Reply:
@viranch That is a good question. Probably this won't be needed (anymore?)

This is old code, see `main` branch:

https://github.com/octodns/octodns-constellix/blob/4566cd358156fc48a32d5aab68f6004160de6fab/octodns_constellix/__init__.py#L1146-L1149

Reply:
@viranch oh, and as an addition 

> every single pool of all dynamic records

Is neither true for the old nor for the modified code. There is
https://github.com/octodns/octodns-constellix/blob/4566cd358156fc48a32d5aab68f6004160de6fab/octodns_constellix/__init__.py#L1135-L1136

above. So the logic seems to be: if a record has _one_ incompatible pool, then the code will disable _all fallbacks for all pools_, but only for this specific "incompatible" record (whatever this is referring too, because the configuration would be valid and not "incompatible", at least with full support for `status`).